### PR TITLE
Added DELTARUNE compatibility

### PR DESCRIPTION
**Pull request by jonloveslegos**
Added DELTARUNE (spelled with all caps)
- Randomizes items, key items, spells, weapons, armor, and starting abilities.

### CODE REVIEW

File: randovania/games/deltarune/patcherfolder/__init__.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            "pickups": [
+                deltarune_pickup_details_to_patcher(detail)
+                for detail in pickup_list
+            ],
+            "starting_items": [
+                deltarune_starting_items_to_patcher(item, qty)
+                for item, qty in patches.starting_items.items()
+            ],
+            "starting_conditions": starting_location_info,
+            "hasSpoiler": description.has_spoiler,
```
Comment (henriquegemignani) [inline randovania/games/deltarune/patcherfolder/__init__.py:177]: `has_spoiler` to be consistent with the rest

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
I'll have some comments about the `patch_game` function later.

**Comment by jonloveslegos** [comment]
I made the requested changes

File: randovania/games/deltarune/patcherfolder/__init__.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            ],
+            "starting_items": [
+                deltarune_starting_items_to_patcher(item, qty)
+                for item, qty in patches.starting_items.items()
+            ],
+            "starting_conditions": starting_location_info,
+            "has_spoiler": description.has_spoiler,
+            "description": "{}".format(description.shareable_word_hash)
+        }
+    def patch_game(self, input_file: Optional[Path], output_file: Path, patch_data: dict,
```
Comment (duncathan) [inline randovania/games/deltarune/patcherfolder/__init__.py:180]: the majority of this function, as well as the `deltapatcher` folder you've added, should probably be in a separate repo and uploaded as a PyPi package for use with pip. if you need assistance getting this set up let us know in discord

**Review by duncathan — CHANGES_REQUESTED** [review]
We've put a few refactors through that affect the game classes since you opened the PR. You'll need to apply those refactors to DELTARUNE when fixing the merge conflicts.

**Comment by duncathan** [comment]
Please resolve the conflicts

File: randovania.spec [hunk 3]
```diff
@@ -22,6 +22,7 @@ a = Analysis(['randovania/__main__.py', 'randovania/cli/__init__.py'],
                  ("randovania/data/hash_words", "data/hash_words"),
                  ("randovania/data/icons", "data/icons"),
                  ("randovania/data/nintendont", "data/nintendont"),
+                 ("deltapatcher", "deltapatcher"),
```
Comment (henriquegemignani) [inline randovania.spec:25]: not needed, as this is handled by the extra package

File: randovania/games/deltarune/game_data.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    gui=_gui,
+
+    generator=game.GameGenerator(
+        item_pool_creator=generator.pool_creator,
+        bootstrap=generator.DeltaruneBootstrap(),
+        base_patches_factory=generator.DeltaruneBasePatchesFactory(),
+    ),
+
+    patcher=patcherfolder.PatcherMaker(),
```
Comment (henriquegemignani) [inline randovania/games/deltarune/game_data.py:45]: This part has changed significantly in newer Randovania versions

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Comment by duncathan** [comment]
it seems that progress on this has stalled. if work continues I will happily reopen the PR!

**Pull request closed without merging**